test/check-machines-migrate

```diff
@@ -0,0 +1,229 @@
+
+    def testFailMigrationAuthentication(self):
+        # No key authentication is set up
+        self._testMigrationGeneric(True, False, "authentication")
+
+    def testFailMigrationUriIncorrect(self):
+        # Incorrect uri inputted
+        self._testMigrationGeneric(True, False, "uri")
+
+    def testFailMigrationDomainUnknown(self):
+        # Destination domain name is not defined in /etc/hosts and is not resolvable
+        self._testMigrationGeneric(True, False, "domain")
+
+    def testFailMigrationStopLibvirtd(self):
+        # Libvirtd.service is not working on the destination
+        self._testMigrationGeneric(True, False, "libvirtd")
+
+    def testFailMigrationKillDestination(self):
+        # Kill destination server in the middle of migration
+        self._testMigrationGeneric(True, False, "kill")
+
+    def _testMigrationGeneric(self, copy_storage, temporary, fail):
+        b = self.browser
+        machine1 = self.machine
+        machine2 = self.machines['machine2']
+
+        self.createVm("subVmTest1")
+        self.login_and_go("/machines")
+        self.prepare(machine1, machine2, fail)
+
+        if not copy_storage:
+            self.preCreateStorageOnDest(machine1, machine2)
+        # Older libvirt needs the destination storage preallocated
+        elif self.machine.image in ["ubuntu-2004"] and fail != "pool":
+            machine2.execute("qemu-img create -f qcow2 /var/lib/libvirt/images/subVmTest1-2.img 50M")
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        self.performAction("subVmTest1", "migrate")
+        b.wait_visible("#migrate-modal")
+
+        if fail == "uri":
+            # test failure for syntactically incorrect destination uri
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/sistem")
+        elif fail == "domain":
+            # test failure for unknown domain
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@someunknowndomain/system")
+        else:
+            b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+
+        if copy_storage:
+            b.click("#copy")
+        else:
+            b.click("#shared")
+
+        b.set_checked("#temporary", temporary)
+        if temporary:
+            b.wait_visible(".pf-c-modal-box__footer .pf-c-alert")
+
+        # The VM does not exist in destination before migration happens
+        self.assertNotIn("subVmTest1", machine2.execute("virsh list --all"))
+
+        if fail == "libvirtd":
+            machine2.execute("systemctl stop libvirtd.service")
+            if machine1.image not in ["debian-stable"]:
+                machine2.execute("systemctl stop libvirtd.socket")
+
+        b.click("#migrate-button")
+        if fail:
+            if fail == "kill":
+                machine2.execute("sleep 1; kill -SEGV 1")
+
+            b.wait_visible(".pf-c-modal-box__footer .pf-c-alert pf-c-alert__title:contains(Migration failed)")
```

Comment:
That [last change](https://github.com/cockpit-project/cockpit-machines/compare/171d6a9af4dab15650b9f629078d823fafe695fa..70d73bface4e9285b3662329218be8c42d947bc4#diff-1745260d862bf4e23c129b75cab71ce43c631d538eb6c26b73b34c97b4ca55aeR191) is probably what [broke all the tests](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-15-20210506-112051-70d73bfa-debian-testing/log.html#55) here?